### ISSUE TITLE
Gutenboarding: improve dark-theme support

### ISSUE THREAD

**Issue by razvanpapadopol** [Automattic/wp-calypso#38587]
Having partial dark-theme support complicates things for development and doesn't provide a consistent experience for the users.

**Current technical concerns:**

- transitioning white -> dark background color and dark -> white text using just opacity is difficult
- style syncing issues between content and background
- trying to support dark-theme only when there is background present (and loaded) seems to be fragile

**User-facing accessibility issues:**

- transition from light to dark theme (once selecting a vertical) and from dark to light (when deselecting a vertical or when advancing to a non-dark step in the onboarding) could be confusing
- we don't have a toggle theme (light/dark) button yet. It should be provided when Gutenberg supports dark-theme

Based on the current findings we consider removing dark preference handling and re-adding it once we have broader support for it (including design).

cc: @dwolfe

![before](https://user-images.githubusercontent.com/841763/71376020-6ebc3c80-25c0-11ea-97ec-7ebe69dcfa22.gif)

**Actions:**

- [x] remove dark-theme support (done in #38771)
- [ ] design a solution to solve issues above or find if it's in progress in Gutenberg
- [ ] re-add dark-theme support to Gutenboarding

**Comment by dwolfe** [comment]
+1 for removing this for now and revisiting later - we'll keep this in mind, and maybe we can sidestep this issue in a future iteration.

**Comment by simison** [comment]
Is this bug with dark/light text fading out oddly on focus also related?

![Screen Recording 2019-12-30 at 15 54 38](https://user-images.githubusercontent.com/87168/71584806-07047500-2b1d-11ea-8ab9-d0710ec41bf8.gif)

Tested with "Digital marketing" topic.

**Comment by razvanpapadopol** [comment]
Yes, that's one of the issues related to dark mode support applied partially only when we have a background. By default in dark mode, we have a dark background and white text so in this case, the placeholder should be white (or a light color) all the time.

**Comment by simison** [comment]
I updated title to "remove" instead of "improve", that's more accurate, right?

**Comment by simison** [comment]
Oh wait, was this already solved at https://github.com/Automattic/wp-calypso/pull/38771 or is there more to this?

**Comment by razvanpapadopol** [comment]
The removing part was indeed solved but the improving part should be kept somewhere in the backlog and it would be helpful to keep this issue open and link new ones to it just to have the full history accessible.
> - [ ] design a solution to solve issues above or find if it's in progress in Gutenberg
> - [ ] re-add dark-theme support to Gutenboarding

Maybe we shouldn't keep it as a _good first issue_ however.

**Comment by simison** [comment]
> design a solution to solve issues above or find if it's in progress in Gutenberg

Gotcha 👍  Yeah we'll need to look at this step design-wise anyway and see how we can improve it.

> re-add dark-theme support to Gutenboarding

I think this has to be Calypso/Gutenberg-wide effort — having just part of the app with dark mode support is valuable excercise in learning but incocnsistent in user experience.

**Comment by simison** [comment]
This needs to come up from the componentry level in the core and rest of Calypso first.
